### ISSUE TITLE
Add support to "parameters" in Python tritonclient package

### ISSUE THREAD

**Issue by RafalSkolasinski** [triton-inference-server/server#4919]
**Is your feature request related to a problem? Please describe.**

I am planning to use Python `tritonclient` for communication with both Triton server and [MLServer](MLServer.readthedocs.io/) (which also implements the V2 inference protocol). In some scenarios I need to be able to set request and inputs `parameters` which seems to currently not be exposed in `tritonclient`.

**Describe the solution you'd like**

1. Extend [InferenceServerClient.infer](https://github.com/triton-inference-server/client/blob/3d05400ed8962c03cb38d45ac4e61276950c3a7a/src/python/library/tritonclient/http/__init__.py#L1347) method to take optional `parameters` argument to be passed to `_get_inference_request` and included in the [infer_request['parameters']](https://github.com/triton-inference-server/client/blob/3d05400ed8962c03cb38d45ac4e61276950c3a7a/src/python/library/tritonclient/http/__init__.py#L109-L110) dict.
2. Extend [InferInput](https://github.com/triton-inference-server/client/blob/3d05400ed8962c03cb38d45ac4e61276950c3a7a/src/python/library/tritonclient/http/__init__.py#L1708) class to add `set_parameters` method that would officially expose adding parameters to `self._parameters` field.

Above example is for HTTP client, ideal solution would be to add similar functionality to also `grpc` and `.aio` version of the clients.

**Describe alternatives you've considered**
I don't see currently any alternative solution.

**Additional context**

**Comment by jbkyang-nvi** [comment]
Is this because you would need your own customized parameters? Are there specific parameters that are not available in the current API?

**Comment by RafalSkolasinski** [comment]
Yes, exactly. We need to set custom parameter called `content_type` to use [codecs](https://mlserver.readthedocs.io/en/latest/user-guide/content-type.html) implemented in MLServer. The [V2 Inference Protocol](https://kserve.github.io/website/modelserving/inference_api) does not seem to specify any limitation on what are the valid parameters.

**Comment by jbkyang-nvi** [comment]
Got it, adding to the team's backlog
